proof/refine/AARCH64/orphanage/Orphanage.thy

```diff
@@ -1144,24 +909,20 @@ proof -
         apply (wp ssa_no_orphans hoare_vcg_all_lift)
             apply (wp hoare_disjI1 chooseThread_nosch)
            apply (wp nextDomain_invs_no_cicd' hoare_vcg_imp_lift
-                     hoare_lift_Pf2 [OF ksQ_all_queued_tcb_ptrs_lift
-                                        [OF nextDomain_ksQ]
-                                        nextDomain_ct']
+                     hoare_lift_Pf2 [OF tcbQueued_all_queued_tcb_ptrs_lift
+                                      [OF nextDomain_tcbQueued]
+                                      nextDomain_ct']
                      hoare_lift_Pf2 [OF st_tcb_at'_is_active_tcb_ptr_lift
                                       [OF nextDomain_st_tcb_at']
                                       nextDomain_ct']
                      hoare_vcg_all_lift getDomainTime_wp)[2]
-          apply ((wp tcbSchedEnqueue_no_orphans tcbSchedEnqueue_in_ksQ'
-                     hoare_drop_imp
-                | clarsimp simp: all_queued_tcb_ptrs_def
-                | strengthen all_invs_but_ct_idle_or_in_cur_domain'_strg
-                | wps tcbSchedEnqueue_ct')+)[1]
-         apply ((wp tcbSchedEnqueue_no_orphans tcbSchedEnqueue_in_ksQ'
-                    hoare_drop_imp
-               | clarsimp simp: all_queued_tcb_ptrs_def
-               | strengthen all_invs_but_ct_idle_or_in_cur_domain'_strg
-               | wps tcbSchedEnqueue_ct')+)[1]
-        apply wp[1]
+          apply wpsimp
+         apply ((wp tcbSchedEnqueue_no_orphans tcbSchedEnqueue_all_queued_tcb_ptrs'
+                      hoare_drop_imp
+                 | clarsimp simp: all_queued_tcb_ptrs_def
+                 | strengthen all_invs_but_ct_idle_or_in_cur_domain'_strg
+                 | rule hoare_lift_Pf2[where f=ksCurThread])+)[1]
```

Comment:
I have a feeling I did try that, but I can check.

Reply:
There's a lot of it around. I wouldn't waste your time diving in and fixing it now necessarily, but I wanted to observe that with `wps` this is in most cases something we might want to avoid in new proofs. The reason is the risk of looping. With the `f=ksCurThread` the risk is reduced (which is how I recognise my own handiwork), but it's still not ideal.